crates/net/network/src/network.rs

```diff
@@ -213,7 +215,17 @@ impl PeersInfo for NetworkHandle {
             let id = *self.peer_id();
             let mut socket_addr = *self.inner.listener_address.lock();
 
-            if socket_addr.ip().is_unspecified() {
+            let external_ip: Option<IpAddr> = self.inner.nat.and_then(|nat| {
```

Comment:
This can just use `self.inner.nat.external_addr()` instead of only supporting the `ExternalIp` variant.